src/vivarium_public_health/metrics/population.py

```diff
@@ -0,0 +1,79 @@
+from collections import Counter
+
+import pandas as pd
+
+from .utilities import get_age_bins, get_population_counts
+
+
+class PopulationObserver:
+    """ An observer for population counts.
+
+    By default, this counts the population at a particular sample date
+    annually. It can be configured to bin the population into age groups and
+    sexes by setting the ``by_age`` and ``by_sex`` flags, respectively. The
+    sample date defaults to July 1st.  This is also configurable
+
+    In the model specification, your configuration for this component should
+    be specified as, e.g.:
+
+    ..code-block: yaml
+
+        configuration:
+            metrics:
+                population:
+                    by_age: True
+                    by_sex: True
+                    sample_date:
+                        month: 4
+                        day: 10
+
+    """
+    configuration_defaults = {
+        'metrics': {
+            'population': {
+                'by_age': False,
+                'by_year': False,
```

Comment:
having by_year here is a little misleading, especially because by_year: False doesn't actually do anything for this observer. If you want to keep it for consistency with the other observers, maybe add a comment indicating that this flag isn't used for this observer/is always overwritten to True in the actual observing?